csrc/host_ir/executor.cpp

```diff
@@ -139,6 +140,11 @@ KernelArgumentHolder HostIrExecutor::run(
         "Output tensor not found in fusion outputs");
     auto out_tensor = output_args[out_idx].as<at::Tensor>();
 
+    inferAndValidateAllocationSizesAndStrides(
```

Comment:
There are other occurrences of postSingleCommunication in this file that are used when host IR lowering is on.

Reply:
After you add validation around those, consider making this a helper so code is less duplicated.